node/hack/governor/src/index.ts

```diff
@@ -290,5 +295,71 @@ axios
     );
   })
   .catch((error) => {
-    console.error(error);
+    console.error("Request error:", error);
   });
+
+
+/*
+  Perform type checks on the incoming values
+  Check for a denylist set of characters
+
+  If either of these fail, we reject adding the token.
+
+  Example data: 30 000000000000000000000000b5c457ddb4ce3312a6c5a2b056a1652bd542a208 O404 omni404 18 1128.69 0xb5c457ddb4ce3312a6c5a2b056a1652bd542a208 7.4832146999999996
+*/
+async function safetyCheck(chain, wormholeAddr, symbol, coinGeckoId, tokenDecimals, tokenPrice, address, notional)  : Promise<boolean>{
+  
+  if(isNaN(chain)){
+    console.log("Invalid chain ID ", chain, " provided")
+    return false; 
+  }
+
+  if(await inputHasInvalidChars(wormholeAddr)){
+    console.log("Invalid wormhole address ", wormholeAddr, " provided")
+    return false; 
+  }
+
+  if(await inputHasInvalidChars(symbol)){
+    console.log("Invalid token symbol ", symbol, " provided")
+    return false; 
+  }
+
+  if(await inputHasInvalidChars(coinGeckoId)){
+    console.log("Invalid coin gecko id ", coinGeckoId, " provided")
+    return false; 
+  }
+
+  if(isNaN(tokenDecimals)){
+    console.log("Invalid token decimals ", tokenDecimals, " provided")
+    return false; 
+  }
+
+  if(isNaN(tokenPrice)){
+    console.log("Invalid token price ", tokenPrice, " provided")
+    return false; 
+  }
+
+  if(await inputHasInvalidChars(address)){
+    console.log("Invalid address ", address, " provided")
+    return false; 
+  }
+  if(isNaN(notional)){
+    console.log("Invalid notional", notional, " provided")
+    return false; 
+  }
+
+  return true; 
+}
+
+// Validates that a list of characters are not in the provided string.
+// If a character is found then return true. Otherwise, return false. 
+async function inputHasInvalidChars(input) : Promise<boolean>{
+  var deny_list = ["\"", "%", "\n","\r", "\\"]
```

Comment:
I think it's worth checking the JSON spec (e.g. here) for valid escape sequences and adding them to the list.

In addition I think we'd be safe to include:
* `/`
* `\`
* `'`
* `"`
* `{`
* `}`
* `[`
* `]`
* `(`
* `)`

Reply:
I think that the `[` and `(` may reasonably used in a token symbol. Since they have no meaning within a string, I'd prefer to keep those out of it. I can add `{}`, `\` and `'` to it though!

Reply:
When I test it locally, we have no governed tokens that use brackets or parentheses. Since this is the case, we might as well add them to the blocklist imo and we can enable them only when we need to later.

The other additions are a nice improvement 👍🏻

Reply:
I know there's nothing there but I could see it happening, was the point. Is there any security issues in Go with not including the brace and paren?

Reply:
I'm not enough of a Go expert to know how to "jailbreak" from the string using them, but since `{}` are used to represent code blocks in Go it might be nice to disable them. Parentheses might be fine?

Reply:
Alright, I just added all of them.